### PR TITLE
Tuples

### PR DESCRIPTION
**Pull request by tuschcarsten**
Tuple types are automatically generated by the compiler, when used.
It is possible to generate classes TupleN for N >= 2. Also Tuple0 with the only value () is possible.
A Tuple1 does not exist, since it would be completely useless.
    
There are three different ways to write tuples and their types:
    
    Tuple2<Int, String> a := Tuple2<Int, String>(1,"Xy")
    println(a._1)
    println(a._2)
    
    Tuple2<Int, String> b := (1,"Xy")
    println(b._1)
    println(b._2)
    
    (Int, String) c := (1,"Xy")
    println(c._1)
    println(c._2)
    
All three notations do the same.
The second one introduces a syntactic sugar for tuple values,
the third one also introduces a syntactic sugar for tuple types.

### CODE REVIEW

**Comment by tuschcarsten** [comment]
performed the refactorings, suggested by @hewie 
- Tuple generation is now implemented in the TypeDeclarationFactory class.
- The generated classe declarations are added to the module declaration after the module is processed.
- static methods are not used
- the class TupleClassDeclaration is now obsolete and was removed.

File: src/main/java/de/uni/bremen/monty/moco/util/TypeDeclarationFactory.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import de.uni.bremen.monty.moco.ast.expression.MemberAccess;
+import de.uni.bremen.monty.moco.ast.expression.SelfExpression;
+import de.uni.bremen.monty.moco.ast.expression.VariableAccess;
+import de.uni.bremen.monty.moco.ast.statement.Assignment;
+import de.uni.bremen.monty.moco.ast.statement.ReturnStatement;
+
+import java.util.*;
+
+/** This class bundles utilities for the automatic generation of types at compile-time */
+public class TypeDeclarationFactory {
```
Comment (hewie) [inline src/main/java/de/uni/bremen/monty/moco/util/TypeDeclarationFactory.java]: Wieso nicht TupleDeclarationFactory, TupleFactory oder sowas?
Momentan ist eh nichts anderes drin und es ist momentan eine übersichtliche, einheitliche Klasse. Wenn man da jetzt noch Logik für was anderes reinpackt (weil der Name das ja hergibt), würde das durchbrochen werden.
Guckst du:
https://en.wikipedia.org/wiki/Single_responsibility_principle 
https://en.wikipedia.org/wiki/God_object

**Comment by tuschcarsten** [comment]
besser?

**Comment by hewie** [comment]
besser!

**Pull request merged**